cmd/gctcli/futures.go

```diff
@@ -1611,3 +1635,78 @@ func setMarginType(c *cli.Context) error {
 	jsonOutput(result)
 	return nil
 }
+
+func getOpenInterest(c *cli.Context) error {
+	if c.NArg() == 0 && c.NumFlags() == 0 {
+		return cli.ShowSubcommandHelp(c)
+	}
+	var (
+		exchangeName, assetType, currencyPair string
+		err                                   error
+	)
+	if c.IsSet("exchange") {
+		exchangeName = c.String("exchange")
+	} else {
+		exchangeName = c.Args().First()
+	}
+
+	if c.IsSet("asset") {
+		assetType = c.String("asset")
+	} else {
+		assetType = c.Args().Get(1)
+	}
+
+	if assetType != "" {
+		err = isFuturesAsset(assetType)
+		if err != nil {
+			return err
+		}
+	}
+
+	if c.IsSet("pair") {
+		currencyPair = c.String("pair")
+	} else {
+		currencyPair = c.Args().Get(2)
+	}
+	var pair currency.Pair
+	if currencyPair != "" {
+		if !validPair(currencyPair) {
+			return fmt.Errorf("%w currencypair:%v", errInvalidPair, currencyPair)
+		}
+		pair, err = currency.NewPairDelimiter(currencyPair, pairDelimiter)
+		if err != nil {
+			return err
+		}
+	}
+
+	data := make([]*gctrpc.OpenInterestDataRequest, 1)
```

Comment:
```suggestion
	data := make([]*gctrpc.OpenInterestDataRequest, 0, 1)
```

And then add append below as this would make it always error on an empty asset as it's len would always be 1 previously

Reply:
Thanks! Fixed